### PR TITLE
[WIP] Fix Intl for PHP >= 7.3 and ICU >= 56

### LINKED ISSUE #163: Tests fail on PHP 7.3
Using PHP 7.3 on Debian unstable, I see the following test failures:
```
There were 18 failures:

1) Symfony\Polyfill\Tests\Apcu\ApcuTest::testApcu
Failed asserting that false is true.

/home/km/projects/symfony-polyfill/tests/Apcu/ApcuTest.php:27

2) Symfony\Polyfill\Tests\Apcu\ApcuTest::testArrayCompatibility
Failed asserting that false matches expected 'value1'.

/home/km/projects/symfony-polyfill/tests/Apcu/ApcuTest.php:52

3) Symfony\Polyfill\Tests\Apcu\ApcuTest::testAPCUIterator
Failed asserting that false is true.

/home/km/projects/symfony-polyfill/tests/Apcu/ApcuTest.php:71

4) Symfony\Polyfill\Tests\Intl\Idn\IdnTest::testDecodeInvalid2003 with data set #0 ('xn--zcaccffbljjkknnoorrssuuxx...g9g.de', 'xn--zcaccffbljjkknnoorrssuuxx...g9g.de')
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'xn--zcaccffbljjkknnoorrssuuxxd5e0a0a3ae9c6a4a9bzdzdxdudwdxd2d2d8d0dse7d6dwe9dxeueweye4eyewe9e5ewkkewc9ftfpfplwexfwf4infvf2f6f6f7f8fpg8fmgngrgrgvgzgygxg3gyg1g3g5gykqg9g.de'
+'äöüßáàăâåãąāæćĉčċçďđéèĕêěëėęēğĝġģĥħíìĭîïĩįīıĵķĺľļłńňñņŋóòŏôőõøōœĸŕřŗśŝšşťţŧúùŭûůűũųūŵýŷÿźžżðþ.de'

/home/km/projects/symfony-polyfill/tests/Intl/Idn/IdnTest.php:83

5) Symfony\Polyfill\Tests\Intl\Idn\IdnTest::testDecodeInvalid2003 with data set #1 ('xn--zcaccffbljjkknnoorrssuuxx...vda.de', 'xn--zcaccffbljjkknnoorrssuuxx...g.þ.de')
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'xn--zcaccffbljjkknnoorrssuuxxd5e0a0a3ae9c8c1b0dxdvdvdxdvd3d0d6dyd8d5d4due7dveseuewe2eweue7e3esk9dxc7frf9e7kuevfuf1ilftf5f4f4f5f6fng6f8f9fpgpgtgxgwgvg1g2gzg1g3gvkog7g.þ.de'
+'äöüßáàăâåãąāæćĉčċçďđéèĕêěëėęēğĝġģĥħíìĭîïĩįīıĵķĺľļłńňñņŋóòŏôőõøōœĸŕřŗśŝšşťţŧúùŭûůűũųūŵýŷÿźžżð.þ.de'

/home/km/projects/symfony-polyfill/tests/Intl/Idn/IdnTest.php:83

6) Symfony\Polyfill\Tests\Intl\Idn\IdnTest::testUppercase2003 with data set #0 ('рф.RU', 'xn--p1ai.RU', 'рф.RU')
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'xn--p1ai.RU'
+'xn--p1ai.ru'

/home/km/projects/symfony-polyfill/tests/Intl/Idn/IdnTest.php:93

7) Symfony\Polyfill\Tests\Intl\Idn\IdnTest::testUppercase2003 with data set #1 ('GUANGDONG.广东', 'GUANGDONG.xn--xhq521b', 'GUANGDONG.广东')
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'GUANGDONG.xn--xhq521b'
+'guangdong.xn--xhq521b'

/home/km/projects/symfony-polyfill/tests/Intl/Idn/IdnTest.php:93

8) Symfony\Polyfill\Tests\Intl\Idn\IdnTest::testUppercase2003 with data set #2 ('renanGonçalves.COM', 'xn--renangonalves-pgb.COM', 'renangonçalves.COM')
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'xn--renangonalves-pgb.COM'
+'xn--renangonalves-pgb.com'

/home/km/projects/symfony-polyfill/tests/Intl/Idn/IdnTest.php:93

9) Symfony\Polyfill\Tests\Intl\Normalizer\NormalizerTest::testConstants
Failed asserting that Array &0 (
    'FORM_C' => 4
    'FORM_D' => 2
    'FORM_KC' => 5
    'FORM_KD' => 3
    'NFC' => 4
    'NFD' => 2
    'NFKC' => 5
    'NFKD' => 3
    'NONE' => 1
) is identical to Array &0 (
    'FORM_C' => 16
    'FORM_D' => 4
    'FORM_KC' => 32
    'FORM_KC_CF' => 48
    'FORM_KD' => 8
    'NFC' => 16
    'NFD' => 4
    'NFKC' => 32
    'NFKC_CF' => 48
    'NFKD' => 8
    'NONE' => 2
).

/home/km/projects/symfony-polyfill/tests/Intl/Normalizer/NormalizerTest.php:37

10) Symfony\Polyfill\Tests\Intl\Normalizer\NormalizerTest::testIsNormalized
Failed asserting that false is true.

/home/km/projects/symfony-polyfill/tests/Intl/Normalizer/NormalizerTest.php:51

11) Symfony\Polyfill\Tests\Intl\Normalizer\NormalizerTest::testNormalize
Failed asserting that false is identical to 'déjà훈쇼™'.

/home/km/projects/symfony-polyfill/tests/Intl/Normalizer/NormalizerTest.php:68

12) Symfony\Polyfill\Tests\Intl\Normalizer\NormalizerTest::testNormalizeConformance
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'Ḋ'
+'Ḋ'

/home/km/projects/symfony-polyfill/tests/Intl/Normalizer/NormalizerTest.php:109

13) Symfony\Polyfill\Tests\Apcu\ApcuTest::testApcu
Polyfills enabled, Failed asserting that false is true.

/home/km/projects/symfony-polyfill/tests/Apcu/ApcuTest.php:27

14) Symfony\Polyfill\Tests\Apcu\ApcuTest::testArrayCompatibility
Polyfills enabled, Failed asserting that false matches expected 'value1'.

/home/km/projects/symfony-polyfill/tests/Apcu/ApcuTest.php:52

15) Symfony\Polyfill\Tests\Apcu\ApcuTest::testAPCUIterator
Polyfills enabled, Failed asserting that false is true.

/home/km/projects/symfony-polyfill/tests/Apcu/ApcuTest.php:71

16) Symfony\Polyfill\Tests\Intl\Normalizer\NormalizerTest::testConstants
Polyfills enabled, Failed asserting that Array &0 (
    'FORM_C' => 4
    'FORM_D' => 2
    'FORM_KC' => 5
    'FORM_KD' => 3
    'NFC' => 4
    'NFD' => 2
    'NFKC' => 5
    'NFKD' => 3
    'NONE' => 1
) is identical to Array &0 (
    'FORM_C' => 16
    'FORM_D' => 4
    'FORM_KC' => 32
    'FORM_KC_CF' => 48
    'FORM_KD' => 8
    'NFC' => 16
    'NFD' => 4
    'NFKC' => 32
    'NFKC_CF' => 48
    'NFKD' => 8
    'NONE' => 2
).

/home/km/projects/symfony-polyfill/tests/Intl/Normalizer/NormalizerTest.php:37

17) Symfony\Polyfill\Tests\Intl\Normalizer\NormalizerTest::testIsNormalized
Polyfills enabled, Failed asserting that true is false.

/home/km/projects/symfony-polyfill/tests/Intl/Normalizer/NormalizerTest.php:53

18) Symfony\Polyfill\Tests\Intl\Normalizer\NormalizerTest::testNormalize
Polyfills enabled, Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'déjà 훈쇼™'
+'déjà 훈쇼™'

/home/km/projects/symfony-polyfill/tests/Intl/Normalizer/NormalizerTest.php:78

FAILURES!
Tests: 1100, Assertions: 364410, Failures: 18, Skipped: 12.

Unsilenced deprecation notices (5)

  5x: Normalizer::NONE is obsolete with ICU 56 and above and will be removed in later PHP versions
    3x in NormalizerTest::testNormalize from Symfony\Polyfill\Tests\Intl\Normalizer
    2x in NormalizerTest::testIsNormalized from Symfony\Polyfill\Tests\Intl\Normalizer

Legacy deprecation notices (136)
```

There's also a Debian bug report at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=911832

### LINKED ISSUE #170: Add Normalizer::FORM_KC_CF
Add in PHP 7.3 as Normalizer::normalize() argument for NFKC_Casefold normalization.

### PR DESCRIPTION
**Pull request by vudaltsov**
Closes #170
Closes #163

Since `PHP 7.3` the behavior of the Intl extension depends on the ICU version ([see this snippet as example](https://github.com/php/php-src/blob/3fa88e0ce0ffd9f63672afe114158a07a0204e21/ext/intl/normalizer/normalizer.h#L22)). 

Thus to be able to test the full range of outcomes and to close #170, we need `PHP 7.3` binary with `ICU >= 56`. Since for Travis CI the ICU version cannot be configured (see https://github.com/travis-ci/travis-ci/issues/3616) and currently is lower than `56`, I decided to compile php manually. The result is cached via standard Travis caching to optimize subsequent builds.

- [x] Make tests fail for `PHP 7.3` and `ICU 63.1`: https://travis-ci.org/symfony/polyfill/jobs/494652921
- [ ] Fix tests
- [ ] Close #170 by implementing the `Normalizer::FORM_KC_CF`

### CODE REVIEW

**Comment by vudaltsov** [comment]
@nicolas-grekas , waiting for your approval of the first step. Ready to work on the rest.

**Comment by nicolas-grekas** [comment]
LGTM thanks!

**Comment by vudaltsov** [comment]
@nicolas-grekas , as you can see in https://travis-ci.org/symfony/polyfill/jobs/495391071, five `@group legacy` tests are still failing. This is strange, looks like `idn_to_utf8` changed it's behavior with `ICU_VERSION=63.1`. Failed to find any explanations. What's the best way to resolve this?

File: src/Intl/Normalizer/Normalizer.php [hunk 1]
```diff
@@ -14,292 +14,61 @@
...
-
-    private static $C;
-    private static $D;
-    private static $KD;
-    private static $cC;
-    private static $ulenMask = array("\xC0" => 2, "\xD0" => 2, "\xE0" => 3, "\xF0" => 4);
-    private static $ASCII = "\x20\x65\x69\x61\x73\x6E\x74\x72\x6F\x6C\x75\x64\x5D\x5B\x63\x6D\x70\x27\x0A\x67\x7C\x68\x76\x2E\x66\x62\x2C\x3A\x3D\x2D\x71\x31\x30\x43\x32\x2A\x79\x78\x29\x28\x4C\x39\x41\x53\x2F\x50\x22\x45\x6A\x4D\x49\x6B\x33\x3E\x35\x54\x3C\x44\x34\x7D\x42\x7B\x38\x46\x77\x52\x36\x37\x55\x47\x4E\x3B\x4A\x7A\x56\x23\x48\x4F\x57\x5F\x26\x21\x4B\x3F\x58\x51\x25\x59\x5C\x09\x5A\x2B\x7E\x5E\x24\x40\x60\x7F\x00\x01\x02\x03\x04\x05\x06\x07\x08\x0B\x0C\x0D\x0E\x0F\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1A\x1B\x1C\x1D\x1E\x1F";
-
-    public static function isNormalized($s, $form = self::NFC)
+if (version_compare(PHP_VERSION, '7.3', '>=') && defined('INTL_ICU_VERSION') && version_compare(INTL_ICU_VERSION, '56', '>=')) {
```
Comment (glensc) [inline src/Intl/Normalizer/Normalizer.php:26]: don't do `version_compare` on `PHP_VERSION`, use compare `PHP_VERSION_ID` instead

**Comment by nicolas-grekas** [comment]
I'm closing because this staled. Please submit again when you're ready!

**Pull request closed without merging**